### PR TITLE
Py junos eznc

### PR DESCRIPTION
**Pull request by xnaveira**
###### Motivation for this change

Some Pypi packages I was missing.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/junos-eznc/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  pname = "junos-eznc";
+  version = "2.1.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "95a037cdd05618a189517357e46a06886909a18c7923b628c6ac43d5f54b2912";
+  };
+
+  buildInputs = [
+    scp six pyserial paramiko netaddr ncclient lxml jinja2 pyyaml
```
Comment (Mic92) [inline pkgs/development/python-modules/junos-eznc/default.nix]: Are all those dependencies only needed at build time of the package? All python dependencies that are needed at runtime, should be also in `propagatedBuildInputs`.

File: pkgs/development/python-modules/ncclient/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "fe6b9c16ed5f1b21f5591da74bfdd91a9bdf69eb4e918f1c06b3c8db307bd32b";
+  };
+
+
+  buildInputs = [
+    setuptools paramiko lxml libxml2 libxslt
+  ];
```
Comment (Mic92) [inline pkgs/development/python-modules/ncclient/default.nix]: same here.

File: pkgs/development/python-modules/scp/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  version = "0.10.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "18f59e48df67fac0b069591609a0f4d50d781a101ddb8ec705f0c2e3501a8386";
+  };
+
+
+  buildInputs = [
+    paramiko
```
Comment (Mic92) [inline pkgs/development/python-modules/scp/default.nix]: same here.

**Comment by xnaveira** [comment]
Ok, changed the dependencies, apparently they are not needed to build.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -452,6 +452,12 @@ in {
 
   python-sybase = callPackage ../development/python-modules/sybase {};
 
+  scp = callPackage ../development/python-modules/scp {};
+
+  ncclient = callPackage ../development/python-modules/ncclient {};
+
+  junos-eznc = callPackage ../development/python-modules/junos-eznc {};
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: The attributes in `python-packages.nix` should be ordered alphabetically.

File: pkgs/development/python-modules/scp/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "18f59e48df67fac0b069591609a0f4d50d781a101ddb8ec705f0c2e3501a8386";
+  };
+
+  propagatedBuildInputs = [
+    paramiko
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/jbardin/scp.py;
+    description = "scp module for paramiko";
```
Comment (dotlambda) [inline pkgs/development/python-modules/scp/default.nix]: Capitalize first letter (or even SCP in this case): https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

File: pkgs/development/python-modules/ncclient/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  version = "0.5.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "fe6b9c16ed5f1b21f5591da74bfdd91a9bdf69eb4e918f1c06b3c8db307bd32b";
+  };
+
+
+  propagatedBuildInputs = [
+    setuptools paramiko lxml libxml2 libxslt
```
Comment (dotlambda) [inline pkgs/development/python-modules/ncclient/default.nix]: `setuptools` should not be needed here

File: pkgs/development/python-modules/scp/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, pythonPackages
+, setuptools
```
Comment (dotlambda) [inline pkgs/development/python-modules/scp/default.nix]: not used

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please ensure that tests are run for each package, if it provides some. You may have to set `checkInputs` and `checkPhase`.
If there are no tests, add the following lines:
```
# no tests implemented
doCheck = false;
```

**Comment by xnaveira** [comment]
Ok, I went through your comments. Thhanks for pinting out the problems.

**Comment by Mic92** [comment]
@GrahamcOfBorg build python36Packages.scp python27Packages.ncclient python27Packages.scp python27Packages.junos-eznc

**Pull request merged**